px-map-tile-layer-wmts.html

```diff
@@ -0,0 +1,185 @@
+@homepage index.html
+@demo index.html
+-->
+
+<script src="leaflet-tilelayer-wmts.js"></script>
+
+<dom-module id="px-map-tile-layer-wmts">
+  <style>
+    :host { display: none }
+  </style>
+  <template></template>
+</dom-module>
+
+<script>
+  Polymer({
+    is: 'px-map-tile-layer-wmts',
+    behaviors: [PxMapBehavior.Layer],
+
+    properties: {
+      /**
+       * Base URL of the WMTS service.
+       */
+      url: {
+        type: String,
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * WMTS layers to show. Required to show the WMTS
+       * tile layers.
+       */
+      layer: {
+        type: String,
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * WMTS request to get the Tile.
+       */
+      request: {
+        type: String,
+        observer: 'shouldUpdateInst',
+        value: 'gettile'
+      },
+
+      /**
+       * Comma-separated list of WMTS styles.
+       */
+      style: {
+        type: String,
+        observer: 'shouldUpdateInst',
+        value: 'default'
+      },
+
+      /**
+       * WMTS image format to use (use 'image/png' for layers).
+       */
+      format: {
+        type: String,
+        value: 'image/jpeg',
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * Version of the WMTS service to use.
+       */
+      version: {
+        type: String,
+        value: '1.0.0',
+        observer: 'shouldUpdateInst'
+      },
+
+      maxNativeZoom: {
+        type: Number,
+        observer: 'shouldUpdateInst',
+        value: 18
+      },
+
+      transparent: {
+        type: Boolean,
+        value: true,
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+       * Tile matrix set to use for the WMTS requests, defaults to
+       * map 'PM'. Don't change this if you're not sure what it means.
+       */
+      tileMatrixSet: {
+        type: String,
+        value: 'default028mm',
+        observer: 'shouldUpdateInst'
+      },
+
+      /**
+      * The identifier of one of the TileMatrix defined in a particular TileMatrixSet
+      */
+      tileMatrix: {
+        type: Number,
+        observer: 'shouldUpdateInst',
+        value: 1
+      },
+
+      /**
+      * Row index of a tile matrix
+      */
+      TileRow: {
+        type: Number,
+        observer: 'shouldUpdateInst',
+        value: 0
+      },
+
+      /**
+      * Column index of a tile matrix
+      */
+      TileCol: {
+        type: Number,
+        observer: 'shouldUpdateInst',
+        value: 0
+      }
+    },
+
+    canAddInst: function() {
+      return (typeof this.url === 'string') && this.url.length > 0 && (typeof this.layer === 'string') && this.layer.length > 0;
+    },
+
+    createInst: function(options) {
+      return L.tileLayer.wmts(options.url, options.wmtsOptions);
+    },
+
+    updateInst: function(lastOptions, nextOptions) {
+      if (lastOptions.url !== nextOptions.url) {
+        this.elementInst.setUrl(nextOptions.url);
```

Comment:
The plugin does not support the `setUrl` method, so you'll need to find another way to do this so the url property can be dynamic. Perhaps the plugin supports updating the instance in another way?

Reply:
Yes the plugin only supports setParams method to update the options, so removed the URL setting. Please let me know if this is correct.